ios/engine/KMEI/KeymanEngine/Classes/Resource Management/ResourceFileManager.swift

```diff
@@ -213,44 +227,52 @@ public class ResourceFileManager {
       throw KMPError.resourceNotInPackage
     }
 
-    // (source, destination)
-    var installableFiles: [(String, URL)] = [(resource.sourceFilename, Storage.active.resourceURL(for: resource)!)]
-    let installableFonts: [(String, URL)] = resource.fonts.map { font in
-      // KMPs only list a single font file for each entry whenever one is included.
-      // A pre-existing assumption.
-      let fontFile = font.source[0]
-      return (fontFile, Storage.active.fontURL(forResource: resource, filename: fontFile)!)
-    }
-
-    installableFiles.append(contentsOf: installableFonts)
-
     do {
-      try FileManager.default.createDirectory(at: Storage.active.resourceDir(for: resource)!,
-                                              withIntermediateDirectories: true)
+      try copyWithOverwrite(from: package.sourceFolder,
+                            to: Storage.active.resourceDir(for: resource)!)
     } catch {
-      log.error("Could not create installation directory: \(error)")
+      log.error("Could not create installation directory and/or copy resources: \(error)")
       throw KMPError.fileSystem
     }
 
-    do {
-      for item in installableFiles {
-        var filePath = package.sourceFolder
-        filePath.appendPathComponent(item.0)
-        try copyWithOverwrite(from: filePath, to: item.1)
+    addResource(resource)
+  }
+
+  internal func addResource<Resource: LanguageResource>(_ resource: Resource) {
+    let path = Storage.active.resourceURL(for: resource)!.path
```

Comment:
This replaces the old `Manager.addKeyboard` and `Manager.lexicalModel` methods, but keeps them framework-internal.

They've been in `Manager` for too long already as it is - `ResourceFileManager` is a more fitting home, as it's the resource-installation core.